common/diagnostics/Diagnostics.java

```diff
@@ -88,40 +88,60 @@ protected Core(Metrics metrics, StatisticReporter statisticReporter, MonitoringS
 
-            Metrics metrics = new Metrics(deploymentID, serverID, distributionName, version, statisticsReportingEnable, dataDirectory);
-            StatisticReporter statisticReporter = initStatisticReporter(deploymentID, statisticsReportingEnable, statisticsReportingURI, metrics, dataDirectory);
-            MonitoringServer monitoringServer = initMonitoringServer(monitoringEnable, monitoringPort, metrics);
+            Metrics metrics =
+                    new Metrics(deploymentID, serverID, distributionName, version, statisticsReportingEnabled, dataDirectory);
+            MonitoringServer monitoringServer = initMonitoringServer(monitoringEnabled, monitoringPort, metrics);
+            StatisticReporter statisticReporter = initStatisticReporter(
+                    developmentModeEnabled, deploymentID, statisticsReportingEnabled, statisticsReportingURI, metrics, dataDirectory);
 
             diagnostics = new Core(metrics, statisticReporter, monitoringServer);
         }
 
         @Nullable
-        protected static MonitoringServer initMonitoringServer(boolean monitoringEnable, int monitoringPort, Metrics metrics) {
-            if (monitoringEnable) return new MonitoringServer(metrics, monitoringPort);
+        protected static MonitoringServer initMonitoringServer(boolean monitoringEnabled, int monitoringPort, Metrics metrics) {
+            if (monitoringEnabled) return new MonitoringServer(metrics, monitoringPort);
             else return null;
         }
 
+        @Nullable
         protected static StatisticReporter initStatisticReporter(
-                String deploymentID, boolean statisticsReportingEnable, String statisticsReportingURI, Metrics metrics, Path dataDirectory
+                boolean developmentModeEnabled,
+                String deploymentID,
+                boolean statisticsReportingEnabled,
+                String statisticsReportingURI,
+                Metrics metrics,
+                Path dataDirectory
         ) {
-            return new StatisticReporter(deploymentID, metrics, statisticsReportingEnable, statisticsReportingURI, dataDirectory);
+            if (!developmentModeEnabled) return new StatisticReporter(
+                    deploymentID, metrics, statisticsReportingEnabled, statisticsReportingURI, dataDirectory);
+            else return null;
         }
 
-        protected static void initSentry(String serverID, String distributionName, String version, boolean errorReportingEnable, String errorReportingURI) {
+        protected static void initSentry(
+                boolean developmentModeEnabled,
+                String serverID,
+                String distributionName,
+                String version,
+                boolean errorReportingEnabled,
+                String errorReportingURI
+        ) {
+            boolean sentryEnabled = !developmentModeEnabled && errorReportingEnabled;
```

Comment:
I was thinking about not even setting usernames, etc for Sentry, just like we do for `Noop`, but decided to just combine these two flags here.